socorro/external/boto/connection_context.py

```diff
@@ -254,29 +233,23 @@ def _convert_list_to_string(self, a_list):
     def _convert_string_to_list(self, a_string):
         return json.loads(a_string)
 
-    def commit(self):
-        """boto doesn't support transactions so this silently
-        does nothing"""
-
-    def rollback(self):
-        """boto doesn't support transactions so this silently
-        does nothing"""
-
     @contextlib.contextmanager
     def __call__(self):
         yield self
 
-    def in_transaction(self, dummy):
-        """boto doesn't support transactions, so it is never in
-        a transaction."""
-        return False
-
     def force_reconnect(self):
-        try:
-            del self.connection
-        except AttributeError:
-            # already deleted, ignorable
-            pass
+        pass
+
+    def is_retryable_exception(self, exc):
+        return (
+            # These are ephemeral problems that might go away with time
+            isinstance(exc, self.RETRYABLE_EXCEPTIONS) or
+
+            # This indicates that the item isn't there, but in the case of
+            # fetching raw crash data, it's possible it's not there *yet* so we
+            # want to give it a few more chances
+            'not found, no value returned' in str(exc)
```

Comment:
Before these changes, this was only checked if the exception was a `StorageResponseError`. As near as I can tell, that can never happened. In the transaction executor, when it was handling the exception, the exception was a `CrashIDNotFound` which was neither in the operational or conditional lists, so it never went through this check.

The problem with the changes I made is that this check triggers when the processor is trying to get the processed crash which doesn't exist in most cases. So this slows down processing for every crash by 5 seconds. Yuck.

I'm going to nix this for now and if we need to re-add some kind of "retry if the crash data isn't there yet" thing, we can do it some other way.